### PR TITLE
Feature/lock unlock actions

### PR DESCRIPTION
**Pull request by eheimbuch**
## Proposed changes

Add lock and unlock action to source view and file content view. If a file is locked it can only be edited by the locking user. But every user is free to remove this lock and edit it again. Also show a warn modal if a user tries to upload and overwrite a locked file.

### Your checklist for this pull request

**Contributor**:
- [x] PR is well described and the description can be used as a commit message on squash
- [x] Related issues linked to PR if existing and labels set
- [x] New code is covered with unit tests
- [ ] New ui components are tested inside the storybook (module ui-components only) 
- [x] [Changelog entry file](https://github.com/scm-manager/changelog#changelog-entry-files) created in `gradle/changelog` or CHANGELOG.md is updated for plugins
- [x] Feature has been tested with different permissions
- [x] Documentation updated (only necessary for new features or changed behaviour)

**Reviewer**:
- [ ] The clean code principles are respected ([CleanCode](https://clean-code-developer.com/virtues/))
- [ ] All new code/logic is implemented on the right spot / "Should this be done here?"
- [ ] UI changes fits into the layout
- [ ] The UI views / components are responsive (mobile views)
- [ ] Correct translations are available

### Checklist for branch merge request (not required for forks)

- [ ] Branch is green/blue on [Jenkins](https://oss.cloudogu.com/jenkins/)
- [ ] Quality Gate passed on [SonarQube](https://sonarcloud.io/organizations/scm-manager/projects)

### CODE REVIEW

File: src/main/js/FileLockUploadModal.tsx [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+        valid = false;
+      }
+    }
+    return valid;
+  };
+
+  const unlockConflictingFiles = () => {
+    const conflictingLocks: FileLock[] = [];
+    for (let file of files) {
+      for (let lock of data?._embedded?.fileLocks as FileLock[]) {
```
Comment (sdorra) [inline src/main/js/FileLockUploadModal.tsx:72]: Iterate over possible undefined?

File: src/main/java/com/cloudogu/filelock/FileLockMapper.java [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+
+  @Named("mapUser")
+  String mapUser(String userId) {
+    return userDisplayManager.get(userId).orElseGet(() -> DisplayUser.from(new User(userId, userId, null))).getDisplayName();
+  }
+
+  @AfterMapping
+  void mapWriteAccess(@MappingTarget FileLockDto dto, FileLock fileLock) {
+    String username = SecurityUtils.getSubject().getPrincipal().toString();
+    dto.setWriteAccess(fileLock.getUserId().equals(username));
```
Comment (sdorra) [inline src/main/java/com/cloudogu/filelock/FileLockMapper.java:79]: I would prefer something like `owned` instead of `writeAccess`

**Review by sdorra — CHANGES_REQUESTED** [review]

**Comment by sdorra** [comment]
The tooltip for a file locked by the current user is misleading. Something like `locked by you at time` would be things more clear.

**Comment by sdorra** [comment]
The download icon has no cursor pointer.

**Comment by sdorra** [comment]
I think the yellow tooltip should describe, that the lock is steeled by clicking the icon.

**Comment by sdorra** [comment]
Wouldn't it be better to remove the file from the upload table after clicking cancel in the upload lock conflict dialog?

**Comment by sdorra** [comment]
Was the download unlock dialog removed?

**Comment by eheimbuch** [comment]
> Wouldn't it be better to remove the file from the upload table after clicking cancel in the upload lock conflict dialog?

And if i select like 10 files? Do i only "reject" the locked ones and keep the unlocked files?

**Comment by eheimbuch** [comment]
> Was the download unlock dialog removed?

Works for me. The lock modal only appears if you download unlocked files.

**Comment by eheimbuch** [comment]
> I think the yellow tooltip should describe, that the lock is steeled by clicking the icon and the green one should say that the file is unlocked after clicking.

Both clicks just unlocks the file. But i will add it :+1:

**Pull request merged**